Fix user-space data write with incorrect offset

RoyWFHuang opened the pull request:
The simplefs_write will copy data from user space. If the
input length is larger than BLOCK_SIZE, the data will not be copied from user space buffer.

"dd" or "print" writes in batches, but "cat" writes the entire data at one.
So we add print and cat commands to test simplefs_write function.

close #66

jserv commented on .github/workflows/main.yaml:
```diff
@@ -22,7 +22,7 @@ jobs:
       - name: coding convention
         run: |
-            sudo apt-get install -q -y clang-format-12
+            sudo apt-get install -q -y clang-format
```
Don't change this. You can draft another pull request for clang-format.

RoyWFHuang replied:
Done, I will create another PR to fix this issue

jserv reviewed (changes requested):
Refine git commit message and append `Close #66` at the end.

jserv commented on script/test.sh:
```diff
@@ -123,6 +123,18 @@ test_op 'dd if=/dev/zero of=file bs=1M count=12 status=none'
 
+# Write the file size larger than BLOCK_SIZE
+# test serial to write
+test_op 'bash -c "printf \"%.0s123456789\" {1..1600} > file.txt"'
```
Why did you use `bash -c`?

RoyWFHuang replied:
Deleted it, sorry it was my mistake to copy my test here

jserv reviewed (changes requested):
Rebase the latest `master` branch.

RoyWFHuang committed:
Fix user-space data write with incorrect offset

	The simplefs_write will copy data from user space. If the
input length is larger than BLOCK_SIZE, the data will not be copied
from user space buffer.

close #66
